src/main/java/ch/njol/skript/expressions/ExprElement.java

```diff
@@ -29,87 +29,148 @@
 import com.google.common.collect.Iterators;
+import org.apache.commons.lang.ArrayUtils;
 import org.bukkit.event.Event;
 import org.eclipse.jdt.annotation.Nullable;
 
 import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Iterator;
-
-@Name("Element of")
-@Description({"The first, last or a random element of a set, e.g. a list variable.",
-		"See also: <a href='#ExprRandom'>random</a>"})
-@Examples("give a random element out of {free items::*} to the player")
-@Since("2.0, INSERT VERSION (relative to last element)")
+import java.util.List;
+
+@Name("Elements")
+@Description({
+		"The first, last, range or a random element of a set, e.g. a list variable.",
+		"See also: <a href='#ExprRandom'>random</a>"
+})
+@Examples("broadcast the first 3 elements of {top players::*}")
+@Since("2.0, INSERT VERSION (relative to last element, range of elements)")
 public class ExprElement extends SimpleExpression<Object> {
 
+	private static final Patterns<ElementType[]> PATTERNS = new Patterns<>(new Object[][]{
+		{"[the] (first|:last) element [out] of %objects%", new ElementType[] {ElementType.FIRST_ELEMENT, ElementType.LAST_ELEMENT}},
+		{"[the] (first|:last) %number% elements [out] of %objects%", new ElementType[] {ElementType.FIRST_X_ELEMENTS, ElementType.LAST_X_ELEMENTS}},
+		{"[a] random element [out] of %objects%", new ElementType[] {ElementType.RANDOM}},
+		{"[the] %number%(st|nd|rd|th) [last:[to] last] element [out] of %objects%", new ElementType[] {ElementType.ORDINAL, ElementType.TAIL_END_ORDINAL}},
+		{"[the] elements (from|between) %number% (to|and) %number% [out] of %objects%", new ElementType[] {ElementType.RANGE}}
+	});
+
 	static {
-		Skript.registerExpression(ExprElement.class, Object.class, ExpressionType.PROPERTY, "(0:[the] first|1:[the] last|2:[a] random|3:[the] %-number%(st|nd|rd|th)|4:[the] %-number%(st|nd|rd|th) [to] last) element [out] of %objects%");
+		Skript.registerExpression(ExprElement.class, Object.class, ExpressionType.PROPERTY, PATTERNS.getPatterns());
 	}
 
 	private enum ElementType {
-		FIRST, LAST, RANDOM, ORDINAL, TAIL_END_ORDINAL
+		FIRST_ELEMENT,
+		LAST_ELEMENT,
+		FIRST_X_ELEMENTS,
+		LAST_X_ELEMENTS,
+		RANDOM,
+		ORDINAL,
+		TAIL_END_ORDINAL,
+		RANGE
 	}
 
-	private ElementType type;
-
 	private Expression<?> expr;
-
-	@Nullable
-	private Expression<Number> number;
+	private	@Nullable Expression<Number> startIndex, endIndex;
+	private ElementType type;
 
 	@Override
 	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		expr = LiteralUtils.defendExpression(exprs[2]);
-		type = ElementType.values()[parseResult.mark];
-		number = (Expression<Number>) (type == ElementType.ORDINAL ? exprs[0]: exprs[1]);
+		ElementType[] types = PATTERNS.getInfo(matchedPattern);
+		expr = LiteralUtils.defendExpression(exprs[exprs.length - 1]);
+		switch (types[0]) {
+			case RANGE:
```

Comment:
shouldn't this also use the parse result?

Reply:
what do you mean?